system-tests/lib/dep-installer/index.ts

```diff
@@ -0,0 +1,310 @@
+import fs from 'fs-extra'
+import path from 'path'
+import cachedir from 'cachedir'
+import execa from 'execa'
+import { cyTmpDir, projectPath, projects, root } from '../fixtures'
+import { getYarnCommand } from './yarn'
+import { getNpmCommand } from './npm'
+
+type Dependencies = Record<string, string>
+
+/**
+ * Type for package.json files for system-tests example projects.
+ */
+type SystemTestPkgJson = {
+  /**
+   * By default, scaffolding will run install if there is a `package.json`.
+   * This option, if set, disables that.
+   */
+  _cySkipDepInstall?: boolean
+  /**
+   * Run the yarn v3-style install command instead of yarn v1-style.
+   */
+  _cyYarnV311?: boolean
+  /**
+   * By default, the automatic install will not run postinstall scripts. This
+   * option, if set, will cause postinstall scripts to run for this project.
+   */
+  _cyRunScripts?: boolean
+  dependencies?: Dependencies
+  devDependencies?: Dependencies
+  optionalDependencies?: Dependencies
+}
+
+const log = (...args) => console.log('📦', ...args)
```

Comment:
Nitpicking here, but a fun TS trick to get the correct type of `...args`:

```ts
const log = (...args: Parameters<typeof console.log>) => console.log('📦', ...args)
```

Reply:
Nice, thanks for the tip. I think I'll leave this one as-is since it's so simple and we use this pattern elsewhere but I'll keep `Parameters` in mind.